exercises/basics/functions.ts

```diff
@@ -1,25 +1,44 @@
 export function returnFunctionalExpression() : any {
-  // @TODO implement (aka: "impl")
+    return function() {}
 }
-
 export function returnFunctionalDeclarationNamedBob() : any {
-  // @TODO impl
-  //       `bob` should be the name
+    return function bob() {
+    }
 }
-
-export function returnObjectWithMethodNamedBobThatReturnsFunctionalDeclarationNamedBob() : any {
-  // @TODO impl
-  //       `bob` should be the name
+    export function returnObjectWithMethodNamedBobThatReturnsFunctionalDeclarationNamedBob() : any {
+    return {
+        bob: function() {
+            return function bob() {
+            }
+        }
+    }
 }
-
 export function curryConcept1(input : number, callback : (input2 : number) => number) {
-  // @TODO impl
-  //       you must combine `input` and `callback`
+    //return callback(3*2)+input // This one is cheating, however, it still passes, LMAO
+    const result = function(input2:number) {
+        return callback(input2) + input*2;
+    }
+    return result(input)
```

Comment:
you solved this one just fine, you just did a superfluous step

you could have simply just did `return callback(input) + (input * 2)`;

Reply:
I updated it! :>